google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -435,48 +515,73 @@ def _schedule_message_on_hold(
         assert self._callback is not None
         self._scheduler.schedule(self._callback, msg)
 
-    def _send_unary_request(self, request: gapic_types.StreamingPullRequest) -> None:
+
+    def send_unary_ack(self, ack_ids, future_reqs_dict) -> tuple[list[requests.AckRequest], list[requests.AckRequest]]:
         """Send a request using a separate unary request instead of over the stream.
 
-        Args:
-            request: The stream request to be mapped into unary requests.
+        If a RetryError occurs, the manager shutdown is triggered, and the
+        error is re-raised.
         """
-        if request.ack_ids:
-            self._client.acknowledge(  # type: ignore
-                subscription=self._subscription, ack_ids=list(request.ack_ids)
+        assert ack_ids
+
+        success = True
+        ack_errors_dict = None
+        try:
+            self._client.acknowledge(
+                subscription=self._subscription, ack_ids=ack_ids
+            )
+        except exceptions.GoogleAPICallError as exc:
+            _LOGGER.debug(
+                "Exception while sending unary RPC. This is typically "
+                "non-fatal as stream requests are best-effort.",
+                exc_info=True,
+            )
+            ack_errors_dict = _get_ack_errors(exc)
+        except exceptions.RetryError as exc:
+            _LOGGER.debug(
+                "RetryError while sending unary RPC. Waiting on a transient "
+                "error resolution for too long, will now trigger shutdown.",
+                exc_info=False,
             )
+            # The underlying channel has been suffering from a retryable error
+            # for too long, time to give up and shut the streaming pull down.
+            self._on_rpc_done(exc)
+            raise
 
-        if request.modify_deadline_ack_ids:
+        requests_completed, requests_to_retry = _process_errors(future_reqs_dict, ack_errors_dict)
```

Comment:
BTW, the name `_process_errors()` is a bit misleading, since we do not actually process any errors if the `acknowledge()` call succeeds.

Reply:
`_process_futures()` is a good choice IMO. :+1: